### PR TITLE
SpeechSynthesisDataset returns `speaker_ids` 

### PR DESCRIPTION
**Pull request by JinZr**

### CODE REVIEW

File: lhotse/dataset/speech_synthesis.py [hunk 1]
```diff
@@ -65,15 +69,21 @@ def __getitem__(self, cuts: CutSet) -> Dict[str, torch.Tensor]:
...
             "audio_lens": audio_lens,
             "features_lens": features_lens,
             "tokens_lens": tokens_lens,
         }
+        if self.speaker_id_mapping is not None:
+            batch["speakers"] = torch.tensor(
+                [self.speaker_id_mapping[cut.supervisions[0].speaker] for cut in cuts],
+                dtype=torch.long,
+            )
+        else:
```
Comment (pzelasko) [inline lhotse/dataset/speech_synthesis.py:85]: This `else:` is a bug, you'd skip `return batch` when `speaker_id_mapping` is available.

File: lhotse/dataset/speech_synthesis.py [hunk 2]
```diff
@@ -82,3 +92,10 @@ def validate_for_tts(cuts: CutSet) -> None:
         assert (
             len(cut.supervisions) == 1
         ), "Only the Cuts with single supervision are supported."
+
+
+def get_sid_to_index_map(sid_list) -> Dict[str, np.ndarray]:
```
Comment (pzelasko) [inline lhotse/dataset/speech_synthesis.py:97]:
```suggestion
def get_sid_to_index_map(sid_list) -> Dict[str, int]:
```

File: lhotse/dataset/speech_synthesis.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 from typing import Callable, Dict, List, Sequence, Union
 
+import numpy as np
```
Comment (pzelasko) [inline lhotse/dataset/speech_synthesis.py:3]: This import won't be needed

**Comment by JinZr** [comment]
we have decided to return the speaker str directly for flexibility, users could make use of the field outside dataloader upon their needs

**Review by pzelasko — APPROVED** [review]

**Pull request merged**